src/cwrapper.h

```diff
@@ -0,0 +1,98 @@
+#ifndef CWRAPPER_H
+#define CWRAPPER_H
+
+#include <gmp.h>
+
+#ifdef __cplusplus
+#include "basic.h"
+#include "constants.h"
+namespace CSymPy {
+    class CWrapper {
+        public:
+            RCP<const Basic> ptr;
+            CWrapper() {
+                ptr = zero;
+            }
+    };
+}
+using CSymPy::CWrapper;
+
+extern "C" {
+
+#else
+typedef struct CWrapper CWrapper;
+#endif
```

Comment:
I thought about that too. Couple things --- try changing the class CWrapper to struct. Maybe that fixes it. The second idea is that maybe a C struct and C++ struct/class cannot be mixed like this. If that's the case, we can just use a `void *` pointer, and just type it to `RCP<Basic>` in each functions.